dialogue-core/src/main/java/com/palantir/dialogue/core/ConcurrencyLimitedChannel.java

```diff
@@ -45,22 +47,35 @@
 
     private final Meter limitedMeter;
     private final LimitedChannel delegate;
-    private final Limiter<Void> limiter;
+    private final SimpleLimiter<Void> limiter;
 
     @VisibleForTesting
-    ConcurrencyLimitedChannel(LimitedChannel delegate, Limiter<Void> limiter, DialogueClientMetrics metrics) {
+    ConcurrencyLimitedChannel(
+            LimitedChannel delegate,
+            SimpleLimiter<Void> limiter,
+            OptionalInt hostIndex,
+            TaggedMetricRegistry taggedMetrics) {
         this.delegate = new NeverThrowLimitedChannel(delegate);
-        this.limitedMeter = metrics.limited(getClass().getSimpleName());
+        this.limitedMeter =
+                DialogueClientMetrics.of(taggedMetrics).limited(getClass().getSimpleName());
         this.limiter = limiter;
+
+        hostIndex.ifPresent(index -> {
+            DialogueConcurrencylimiterMetrics metrics = DialogueConcurrencylimiterMetrics.of(taggedMetrics);
+            // TODO(dfox): hook up the service-name somehow? also when nodes reshuffle these metrics will look odd.
+            metrics.utilization().hostIndex(Integer.toString(index)).build(this::getUtilization);
```

Comment:
I'm not confident this works when we have multiple discovered services, each combined channel (alta both gatekeeper, for example) will have separate concurrency limited channels using the same gauge for index zero. We don't know which one we're measuring.

Reply:
Yep that's why I added the `do not merge`. I think the thing we're really interested in here is some stable identifier per host.

I don't think we can actually use hashcode because then a ton of uri reloads will result in an unbounded number of tags :(